cppcache/src/TcrMessage.cpp

```diff
@@ -1373,14 +1360,10 @@ void TcrMessage::handleByteArrayResponse(
 
     case TcrMessage::LOCAL_CREATE:
     case TcrMessage::LOCAL_UPDATE: {
-      int32_t regionLen = input.readInt32();
+      m_regionName.resize(input.readInt32());
       input.advanceCursor(1);  // ignore byte
-      char* regname = nullptr;
-      regname = new char[regionLen + 1];
-      DeleteArray<char> delRegName(regname);
-      input.readBytesOnly(reinterpret_cast<int8_t*>(regname), regionLen);
-      regname[regionLen] = '\0';
-      m_regionName = regname;
+      input.readBytesOnly(reinterpret_cast<int8_t*>(&m_regionName[0]),
```

Comment:
Seems very unsafe! Do we know that all implementations of `std::string` have allocated a contiguous space large enough to do this? There is a reason that `c_str()` and `data()` are `const`.  Is the null termination correctly handled here?